src/ssmDocument/commands/openDocumentItem.ts

```diff
@@ -0,0 +1,134 @@
+        const rawContent = await node.getDocumentContent(documentVersion, documentFormat)
+        const textDocument = await vscode.workspace.openTextDocument({
+            content: rawContent.Content,
+            language: `ssm-${rawContent.DocumentFormat!.toLowerCase()}`,
+        })
+        vscode.window.showTextDocument(textDocument)
+    } catch (err) {
+        result = 'Failed'
+        const error = err as Error
+        vscode.window.showErrorMessage(
+            localize(
+                'AWS.message.error.ssmDocumet.openDocument.could_not_open',
+                'Could not fetch and display document {0} contents',
+                node.documentName
+            )
+        )
+        logger.error('Error on openning document: %0', error)
+    } finally {
+        telemetry.recordSsmOpenDocument({ result: result })
+    }
+}
+
+async function promptUserforDocumentFormat(formats: string[]): Promise<string | undefined> {
+    // Prompt user to pick document format
+    const quickPickItems: SsmDocumentQuickPickItem[] = formats.map(format => {
+        return {
+            label: format,
+            description: `Open document with format ${format}`,
+        }
+    })
+
+    const formatPick = picker.createQuickPick({
+        options: {
+            ignoreFocusOut: true,
+            title: localize('AWS.message.prompt.selectSsmDocumentFormat.placeholder', 'Select a document format'),
+        },
+        buttons: [vscode.QuickInputButtons.Back],
+        items: quickPickItems,
+    })
+
+    const formatChoices = await picker.promptUser({
+        picker: formatPick,
+        onDidTriggerButton: (_, resolve) => {
+            resolve(undefined)
+        },
+    })
+
+    const formatSelection = picker.verifySinglePickerOutput(formatChoices)
+
+    // User pressed escape and didn't select a template
+    if (formatSelection === undefined) {
+        return undefined
+    }
+
+    return formatSelection.label
+}
+
+async function promptUserforDocumentVersion(versions: SSM.Types.DocumentVersionInfo[]): Promise<string | undefined> {
+    // Prompt user to pick document version
+    const quickPickItems: SsmDocumentQuickPickItem[] = []
+    versions.forEach(version => {
+        if (version.DocumentVersion) {
+            quickPickItems.push({
+                label: version.DocumentVersion,
+                description: `${version.IsDefaultVersion ? 'Default' : ''}`,
+            })
+        }
+    })
+
+    if (quickPickItems.length > 1) {
+        const versionPick = picker.createQuickPick({
+            options: {
+                ignoreFocusOut: true,
+                title: localize('AWS.message.prompt.selectSsmDocumentVersion.placeholder', 'Select a document version'),
+            },
+            buttons: [vscode.QuickInputButtons.Back],
+            items: quickPickItems,
+        })
+
+        const versionChoices = await picker.promptUser({
+            picker: versionPick,
+            onDidTriggerButton: (_, resolve) => {
```

Comment:
Question: Are you actually handling the back input (same in other quick pick cases)? It doesn't look like it...For reference, this is a pattern we use in our wizards, and we handle it by explicitly going back a step when the user selects an undefined value. For instance, [this back input](https://github.com/aws/aws-toolkit-vscode/blob/12e5c2e28621b384834286e1021dbef17a0e206d/src/lambda/wizards/samInitWizard.ts#L110) bubbles up to [this statement](https://github.com/aws/aws-toolkit-vscode/blob/12e5c2e28621b384834286e1021dbef17a0e206d/src/lambda/wizards/samInitWizard.ts#L384), which on `undefined` returns to the previous step. If you don't need it, remove the `buttons` field from the quick pick; it's not mandatory.

Nit: I don't like naming the button param for the callback `_`, especially when we frequently use `_` in our codebase for `lodash`.

Reply:
Will remove buttons and onDidTriggerButton